anesthetic/utils.py

```diff
@@ -210,17 +209,43 @@ def iso_probability_contours_from_samples(pdf, contours=[0.68, 0.95],
     return c
 
 
-def triangular_sample_compression_2d(x, y, w=None, n=1000):
+def scaled_triangulation(x, y, cov):
+    """Triangulation scaled by a covariance matrix.
+
+    Parameters
+    ----------
+    x, y: array-like
+        x and y coordinates of samples
+
+    cov: array-like, 2d
+        Covariance matrix for scaling
+
+    Returns
+    -------
+    matplotlib.tri.Triangulation
+        Triangulation with the appropriate scaling
+    """
+    L = numpy.linalg.cholesky(cov)
+    Linv = numpy.linalg.inv(L)
+    x_, y_ = Linv.dot([x, y])
+    tri = Triangulation(x_, y_)
+    return Triangulation(x, y, tri.triangles)
+
+
+def triangular_sample_compression_2d(x, y, cov, w=None, n=1000):
     """Histogram a 2D set of weighted samples via triangulation.
 
-    This defines bins via a triangulation of the subsamples, sums weights
-    within triangles, and computes weighted centroids of triangles.
+    This defines bins via a triangulation of the subsamples and sums weights
+    within triangles surrounding point each point
```

Comment:
`point each point`

typo?

Reply:
Fixed, and also added test to check this function behaves as appropriate